classes/PaymentModule.php

```diff
@@ -1274,18 +1307,211 @@ protected function createOrderCartRules(
         return $cart_rules_list;
     }
 
-    private function addShipmentToOrder(Order $order)
-    {
-        /** @var FeatureFlagStateCheckerInterface $featureFlagManager */
-        $featureFlagManager = $this->get(FeatureFlagStateCheckerInterface::class);
+    private function createOrderFromCartForMultiShipment(
```

Comment:
Isn't it overkill to duplicate almost the whole inital method Maybe my suggestion were confusing but the purpose was to reduce code duplication and it the end it ends up with with even more code duplicated 🤔

Isn't it possible for `createOrderFromCartForMultiShipment` to call `createOrderFromCart` instead, it's just that it is responsible for formatting the data differently compared to when the feature flag is not enabled